drivers/gles3/rasterizer_scene_gles3.cpp

```diff
@@ -5105,7 +5105,7 @@ void RasterizerSceneGLES3::initialize() {
 
 	shadow_atlas_realloc_tolerance_msec = 500;
 
-	int max_shadow_cubemap_sampler_size = 512;
+	int max_shadow_cubemap_sampler_size = CLAMP(int(next_power_of_2(GLOBAL_GET("rendering/quality/shadow_atlas/size")) << 2), 512, 2048);
```

Comment:
The logic here is questionable. 

At the default shadow atlas size of 4096, this equation returns ``CLAMP(16384, 512, 2048)`` which clamps to 2048. with a shadow atlas size of 512 we still end up with ``CLAMP(2048, 512, 2048)`` which returns 2048. 

Later in the pipeline, Godot is able to select the appropriate cubemap size. But the largest cubemap size we need is the size of the shadow atlas. This logic would result in us allocating a substantially larger buffer than needed (and thus wasting VRAM

It wouldn't make sense to render to a 2048x2048 cubemap only to copy that cubemap into a 256x256 portion of the 512x512 shadow atlas

Reply:
Oh, I see the operator was backwards!... This should be fixed now. Thanks

The idea was to have the default 4096 still use the old hardcoded 512 (as originally intended)

I lowered the minimum to 256, too.

Reply:
To get to 512 from 4096, you would have to bitshift by 3 i.e. ``4096 >> 3 = 512``